covalent_dispatcher/_core/execution.py

```diff
@@ -281,9 +283,48 @@ def _run_planned_workflow(result_object: Result, thread_pool: ThreadPoolExecutor
         None
     """
 
+    tasks_queue = Queue()
+    pending_deps = {}
+    tasks_left = 0
+    lock = Lock()
+    futures: list = []
+
     def update_node_result(node_result: dict):
+        with lock:
+            result_object._update_node(**node_result)
+            with DispatchDB() as db:
+                db.upsert(result_object.dispatch_id, result_object)
+                db.save_db(result_object)
+            result_object.save()
+            result_webhook.send_update(result_object)
+            g = result_object.lattice.transport_graph._graph
+            if node_result["status"] == Result.COMPLETED:
+                for child in g.neighbors(node_result["node_id"]):
+                    pending_deps[child] -= 1
```

Comment:
I overlooked the case where there are multiple edges between two nodes. We should have something like
```python
degree = len(g.edges( node_result["node_id"], child ))
pending_deps[child] -= degree
```